File: decompiled-improved/io/legado/app/utils/JsoupExtensionsKt.java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  kotlin.Metadata
 *  kotlin.jvm.internal.Intrinsics
 *  org.jetbrains.annotations.NotNull
 *  org.jsoup.internal.StringUtil
 *  org.jsoup.nodes.CDataNode
 *  org.jsoup.nodes.Element
 *  org.jsoup.nodes.Node
 *  org.jsoup.nodes.TextNode
 *  org.jsoup.select.NodeTraversor
 *  org.jsoup.select.NodeVisitor
 */
package io.legado.app.utils;

import io.legado.app.utils.StringExtensionsKt;
import kotlin.Metadata;
import kotlin.jvm.internal.Intrinsics;
import org.jetbrains.annotations.NotNull;
import org.jsoup.internal.StringUtil;
import org.jsoup.nodes.CDataNode;
import org.jsoup.nodes.Element;
import org.jsoup.nodes.Node;
import org.jsoup.nodes.TextNode;
import org.jsoup.select.NodeTraversor;
import org.jsoup.select.NodeVisitor;

@Metadata(mv={1, 5, 1}, k=2, xi=48, d1={"\u00006\n\u0000\n\u0002\u0010\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\u000b\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\u0011\n\u0002\u0010\u000e\n\u0002\u0018\u0002\n\u0002\b\u0002\u001a\u001c\u0010\u0000\u001a\u00020\u00012\n\u0010\u0002\u001a\u00060\u0003j\u0002`\u00042\u0006\u0010\u0005\u001a\u00020\u0006H\u0002\u001a\u0010\u0010\u0007\u001a\u00020\b2\u0006\u0010\u0002\u001a\u00020\u0003H\u0002\u001a\u0012\u0010\t\u001a\u00020\b2\b\u0010\n\u001a\u0004\u0018\u00010\u000bH\u0002\u001a\u0015\u0010\f\u001a\b\u0012\u0004\u0012\u00020\u000e0\r*\u00020\u000f\u00a2\u0006\u0002\u0010\u0010\u00a8\u0006\u0011"}, d2={"appendNormalisedText", "", "sb", "Ljava/lang/StringBuilder;", "Lkotlin/text/StringBuilder;", "textNode", "Lorg/jsoup/nodes/TextNode;", "lastCharIsWhitespace", "", "preserveWhitespace", "node", "Lorg/jsoup/nodes/Node;", "textArray", "", "", "Lorg/jsoup/nodes/Element;", "(Lorg/jsoup/nodes/Element;)[Ljava/lang/String;", "reader-pro"})
public final class JsoupExtensionsKt {
    /*
     * WARNING - void declaration
     */
    @NotNull
    public static final String[] textArray(@NotNull Element $this$textArray) {
        void $this$trim$iv;
        Intrinsics.checkNotNullParameter((Object)$this$textArray, (String)"<this>");
        StringBuilder sb = StringUtil.borrowBuilder();
        NodeTraversor.traverse((NodeVisitor)new NodeVisitor(sb){
            final /* synthetic */ StringBuilder $sb;
            {
                this.$sb = $sb;
            }

            public void head(@NotNull Node node, int depth) {
                Intrinsics.checkNotNullParameter((Object)node, (String)"node");
                if (node instanceof TextNode) {
                    StringBuilder stringBuilder = this.$sb;
                    Intrinsics.checkNotNullExpressionValue((Object)stringBuilder, (String)"sb");
                    JsoupExtensionsKt.access$appendNormalisedText(stringBuilder, (TextNode)node);
                } else if (node instanceof Element) {
                    CharSequence charSequence = this.$sb;
                    Intrinsics.checkNotNullExpressionValue((Object)charSequence, (String)"sb");
                    charSequence = charSequence;
                    boolean bl = false;
                    if (charSequence.length() > 0 && (((Element)node).isBlock() || Intrinsics.areEqual((Object)((Element)node).tag().getName(), (Object)"br"))) {
                        charSequence = this.$sb;
                        Intrinsics.checkNotNullExpressionValue((Object)charSequence, (String)"sb");
                        if (!JsoupExtensionsKt.access$lastCharIsWhitespace((StringBuilder)charSequence)) {
                            this.$sb.append("\n");
                        }
                    }
                }
            }

            public void tail(@NotNull Node node, int depth) {
                Intrinsics.checkNotNullParameter((Object)node, (String)"node");
                if (node instanceof Element && ((Element)node).isBlock() && node.nextSibling() instanceof TextNode) {
                    StringBuilder stringBuilder = this.$sb;
                    Intrinsics.checkNotNullExpressionValue((Object)stringBuilder, (String)"sb");
                    if (!JsoupExtensionsKt.access$lastCharIsWhitespace(stringBuilder)) {
                        this.$sb.append("\n");
                    }
                }
            }
        }, (Node)((Node)$this$textArray));
        String[] stringArray = StringUtil.releaseBuilder((StringBuilder)sb);
        Intrinsics.checkNotNullExpressionValue((Object)stringArray, (String)"releaseBuilder(sb)");
        boolean $i$f$trim = false;
        CharSequence $this$trim$iv$iv = (CharSequence)$this$trim$iv;
        boolean $i$f$trim2 = false;
        int startIndex$iv$iv = 0;
        int endIndex$iv$iv = $this$trim$iv$iv.length() - 1;
        boolean startFound$iv$iv = false;
        while (startIndex$iv$iv <= endIndex$iv$iv) {
            boolean match$iv$iv;
            int index$iv$iv = !startFound$iv$iv ? startIndex$iv$iv : endIndex$iv$iv;
            char it = $this$trim$iv$iv.charAt(index$iv$iv);
            boolean bl = false;
            boolean bl2 = match$iv$iv = Intrinsics.compare((int)it, (int)32) <= 0;
            if (!startFound$iv$iv) {
                if (!match$iv$iv) {
                    startFound$iv$iv = true;
                    continue;
                }
                ++startIndex$iv$iv;
                continue;
            }
            if (!match$iv$iv) break;
            --endIndex$iv$iv;
        }
        String text = ((Object)$this$trim$iv$iv.subSequence(startIndex$iv$iv, endIndex$iv$iv + 1)).toString();
        stringArray = new String[]{"\n"};
        return StringExtensionsKt.splitNotBlank(text, stringArray);
    }

    private static final void appendNormalisedText(StringBuilder sb, TextNode textNode) {
        String text = textNode.getWholeText();
        if (JsoupExtensionsKt.preserveWhitespace(textNode.parentNode()) || textNode instanceof CDataNode) {
            sb.append(text);
        } else {
            StringUtil.appendNormalisedWhitespace((StringBuilder)sb, (String)text, (boolean)JsoupExtensionsKt.lastCharIsWhitespace(sb));
        }
    }

    private static final boolean preserveWhitespace(Node node) {
        if (node instanceof Element) {
            int n;
            Element el = (Element)node;
            int i = 0;
            do {
                if (el.tag().preserveWhitespace()) {
                    return true;
                }
                el = el.parent();
            } while ((i = (n = i) + 1) < 6 && el != null);
        }
        return false;
    }

    private static final boolean lastCharIsWhitespace(StringBuilder sb) {
        CharSequence charSequence = sb;
        boolean bl = false;
        return charSequence.length() > 0 && sb.charAt(sb.length() - 1) == ' ';
    }

    public static final /* synthetic */ void access$appendNormalisedText(StringBuilder sb, TextNode textNode) {
        JsoupExtensionsKt.appendNormalisedText(sb, textNode);
    }

    public static final /* synthetic */ boolean access$lastCharIsWhitespace(StringBuilder sb) {
        return JsoupExtensionsKt.lastCharIsWhitespace(sb);
    }
}

File: full-vineflower/io/legado/app/utils/JsoupExtensionsKt.kt
package io.legado.app.utils

import kotlin.jvm.internal.Intrinsics
import org.jetbrains.annotations.NotNull
import org.jsoup.internal.StringUtil
import org.jsoup.nodes.CDataNode
import org.jsoup.nodes.Element
import org.jsoup.nodes.Node
import org.jsoup.nodes.TextNode
import org.jsoup.select.NodeTraversor
import org.jsoup.select.NodeVisitor

public fun Element.textArray(): Array<String> {
   val sb: StringBuilder = StringUtil.borrowBuilder();
   NodeTraversor.traverse(new NodeVisitor(sb) {
      {
         this.$sb = `$sb`;
      }

      @Override
      public void head(@NotNull Node node, int depth) {
         if (node is TextNode) {
            val var3: StringBuilder = this.$sb;
            JsoupExtensionsKt.access$appendNormalisedText(var3, node as TextNode);
         } else if (node is Element) {
            var var5: StringBuilder = this.$sb;
            if (var5.length() > 0 && ((node as Element).isBlock() || (node as Element).tag().getName() == "br")) {
               var5 = this.$sb;
               if (!JsoupExtensionsKt.access$lastCharIsWhitespace(var5)) {
                  this.$sb.append("\n");
               }
            }
         }
      }

      @Override
      public void tail(@NotNull Node node, int depth) {
         if (node is Element && (node as Element).isBlock() && node.nextSibling() is TextNode) {
            val var3: StringBuilder = this.$sb;
            if (!JsoupExtensionsKt.access$lastCharIsWhitespace(var3)) {
               this.$sb.append("\n");
            }
         }
      }
   }, `$this$textArray`);
   val `$this$trim$iv`: java.lang.String = StringUtil.releaseBuilder(sb);
   val `$this$trim$iv$iv`: java.lang.CharSequence = `$this$trim$iv`;
   var `startIndex$iv$iv`: Int = 0;
   var `endIndex$iv$iv`: Int = `$this$trim$iv$iv`.length() - 1;
   var `startFound$iv$iv`: Boolean = false;

   while (startIndex$iv$iv <= endIndex$iv$iv) {
      val var14: Boolean = Intrinsics.compare(`$this$trim$iv$iv`.charAt(if (!`startFound$iv$iv`) `startIndex$iv$iv` else `endIndex$iv$iv`), 32) <= 0;
      if (!`startFound$iv$iv`) {
         if (!var14) {
            `startFound$iv$iv` = true;
         } else {
            `startIndex$iv$iv`++;
         }
      } else {
         if (!var14) {
            break;
         }

         `endIndex$iv$iv`--;
      }
   }

   return StringExtensionsKt.splitNotBlank(`$this$trim$iv$iv`.subSequence(`startIndex$iv$iv`, `endIndex$iv$iv` + 1).toString(), "\n");
}

private fun appendNormalisedText(sb: StringBuilder, textNode: TextNode) {
   val text: java.lang.String = textNode.getWholeText();
   if (!preserveWhitespace(textNode.parentNode()) && textNode !is CDataNode) {
      StringUtil.appendNormalisedWhitespace(sb, text, lastCharIsWhitespace(sb));
   } else {
      sb.append(text);
   }
}

private fun preserveWhitespace(node: Node?): Boolean {
   if (node is Element) {
      var el: Element = node as Element;
      val i: Int = 0;

      do {
         if (el.tag().preserveWhitespace()) {
            return true;
         }

         el = el.parent();
      } while (++i < 6 && el != null);
   }

   return false;
}

private fun lastCharIsWhitespace(sb: StringBuilder): Boolean {
   return sb.length() > 0 && sb.charAt(sb.length() - 1) == ' ';
}

@JvmSynthetic
fun `access$appendNormalisedText`(sb: StringBuilder, textNode: TextNode) {
   appendNormalisedText(sb, textNode);
}

@JvmSynthetic
fun `access$lastCharIsWhitespace`(sb: StringBuilder): Boolean {
   return lastCharIsWhitespace(sb);
}
